generator/ServiceClientGeneratorLib/Generators/AuthResolvers/ModeledResolver.tt

```diff
@@ -35,14 +35,33 @@ namespace <#=this.Config.Namespace#>.Internal
     /// </summary>
     public class Amazon<#=this.Config.ClassName#>AuthSchemeHandler : BaseAuthResolverHandler
     {
+<# if (IsServiceAllowListedForRulesBasedResolver()) { #>
+        private readonly Amazon<#=this.Config.ClassName#>EndpointResolver _endpointResolver = new();
+
+<# } #>
         /// <summary>
-        /// Auth scheme resolver for AutoScaling.
+        /// Modeled auth scheme resolver for <#=this.Config.ClassName#>.
         /// </summary>
         public Amazon<#=this.Config.ClassName#>AuthSchemeResolver AuthSchemeResolver { get; } = new();
 
         /// <inheritdoc/>
-        protected override List<IAuthSchemeOption> ResolveAuthOptions(IRequestContext requestContext)
+        protected override List<IAuthSchemeOption> ResolveAuthOptions(IExecutionContext executionContext)
```

Comment:
I wanted to get feedback on this approach I took: The SRA docs mention we _should code generate the mapping from auth scheme parameters to endpoint resolver parameters_, however that was a lot of duplication for S3. By calling the generated resolver instead, we'll use all endpoint parameters anyways (including new ones we might add in the future): https://github.com/aws/aws-sdk-net/blob/36af0432b27eda1efb55c8f3ed6c8c11326f7ddd/sdk/src/Services/S3/Generated/Internal/AmazonS3EndpointResolver.cs#L70-L82

One benefit is that it allowed me to keep the implementation to retrieve auth schemes from an endpoint (in `sdk/src/Core/Amazon.Runtime/Pipeline/Handlers/BaseAuthResolverHandler.cs`) simpler.

Reply:
How would rerunning the endpoint resolver twice work when the SDK supports account id based endpoint rules? Because we would need the identity to get the account id for the endpoint rules I believe.

Reply:
If we do need to run endpoint rules twice and it will use the exact same parameters we could stash the results in `executionContext.RequestContext.ContextAttributes` and then have the endpoint resolver check for the existence and if so reuse those results.

Reply:
In the internal design doc (that recommended running the EP2.0 resolver twice), there was a comment on not allowing S3 to use the `AWS::Auth::AccountId` parameter in rules that have authentication schemes.

I'll follow up to see if that was ever built or if it's something we should do in our generator.

Reply:
For the other comment, there's already an `EndpointAttributes` property in the request context: https://github.com/aws/aws-sdk-net/blob/ad13fd46500b624f3e39190a8ed9f32633ca0559/sdk/src/Core/Amazon.Runtime/Internal/IRequest.cs#L385-L388

I guess we could populate that? (although the same concern with running the resolver twice applies). That property is used in other S3 classes to check if it's running on S3 Express: https://github.com/aws/aws-sdk-net/blob/ad13fd46500b624f3e39190a8ed9f32633ca0559/sdk/src/Services/S3/Custom/Util/AmazonS3Util.cs#L371-L378

Reply:
For the other comment, I thought about populating the existing `EndpointAttributes` property: https://github.com/aws/aws-sdk-net/blob/ad13fd46500b624f3e39190a8ed9f32633ca0559/sdk/src/Core/Amazon.Runtime/Internal/IRequest.cs#L388

But it looks like when the endpoints resolver calls `GetEndpoint` it calculates them again anyway (this constructor is called by the generated code): https://github.com/aws/aws-sdk-net/blob/ad13fd46500b624f3e39190a8ed9f32633ca0559/sdk/src/Core/Amazon.Runtime/Endpoints/Endpoint.cs#L45-L49

Or did you mean I should cache the entire endpoint? (I don't think that's the case since we'd need to change whenever we add support for account-ID based endpoints).

Reply:
My understanding was running the endpoint rules twice would only work if we the service didn't use account id as part of the rule. In that case would we need to change this when the SDK supports account id based rules? So yes I was suggesting caching the results within the scope of the current request in stead of running the endpoint rules twice. But maybe I'm misunderstanding something and the second time could cause a different results in some scenarios.